src/evaluate/evaluation_suite/__init__.py

```diff
@@ -0,0 +1,98 @@
+import importlib
+import inspect
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Callable, Optional, Union
+
+from datasets import Dataset, DownloadMode, load_dataset
+from datasets.utils.version import Version
+
+from evaluate.evaluator import evaluator
+from evaluate.loading import evaluation_module_factory
+from evaluate.utils.file_utils import DownloadConfig
+from evaluate.utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class SubTask:
+    task_type: str
+    data: Optional[Union[str, Dataset]] = None
+    subset: Optional[str] = None
+    split: Optional[str] = None
+    data_preprocessor: Optional[Callable] = None
+    args_for_task: Optional[dict] = None
+
+
+def import_main_class(module_path):
+    """Import a module at module_path and return the EvaluationSuite class"""
+    module = importlib.import_module(module_path)
+
+    module_main_cls = None
+    for name, obj in module.__dict__.items():
+        if isinstance(obj, type) and obj.__name__ == "Suite":
+            if inspect.isabstract(obj):
+                continue
+            module_main_cls = obj
+            break
+
+    return module_main_cls
+
+
+def load(
+    path,
+    download_mode: Optional[DownloadMode] = None,
+    revision: Optional[Union[str, Version]] = None,
+    download_config: Optional[DownloadConfig] = None,
+):
+    download_mode = DownloadMode(download_mode or DownloadMode.REUSE_DATASET_IF_EXISTS)
+    evaluation_module = evaluation_module_factory(
+        path, module_type=None, revision=revision, download_config=download_config, download_mode=download_mode
+    )
+    name = Path(path).stem
+    evaluation_cls = import_main_class(evaluation_module.module_path)
+    evaluation_instance = evaluation_cls(name)
+
+    return evaluation_instance
+
+
+class EvaluationSuite:
+    """
+    This class instantiates an evaluation suite made up of multiple tasks, where each task consists of a dataset and
+    an associated metric, and runs evaluation on a model or pipeline. Evaluation suites with a Python script found
+    either locally or uploaded as a dataset on the Hugging Face Hub.
+    Usage:
+    ```python
+    from evaluate.evaluation_suite import load_evaluation_suite
+    suite = load_evaluation_suite('mathemakitten/glue-suite-v2')
+    results = suite.run("gpt2")
+    ```
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+    def run(self, model_or_pipeline=None):
+
+        self.setup()
+
+        results_all = {}
+        for task in self.suite:
+
+            if task.data_preprocessor:  # task requires extra preprocessing
+                ds = load_dataset(task.data, split=task.split)
```

Comment:
In `datasets` we call this a "configuration name" btw. Sometimes it is as subset, sometimes it has another meaning

Reply:
Adding, thanks for catching! I'm thinking like `ds = load_dataset(task.data, name=task.subset, split=task.split)` so it matches up with the evaluator convention used here, which uses `datasets`'s `name` under the hood: https://github.com/huggingface/evaluate/pull/283/files

Mostly just because having "name" as a field outside of the `datasets` library is somewhat ambiguous for the user (i.e. it's sometimes unclear if `name` is used standalone whether it's referring to a dataset name or the name of some other object)